robot_smach_states/benchmark/benchmark_grasping.py

```diff
@@ -0,0 +1,105 @@
+#! /usr/bin/env python
+
+# System
+import argparse
+import csv
+import time
+
+# ROS
+import rospy
+
+# TU/e Robotics
+from robot_skills.get_robot import get_robot
+
+# Robot Smach States
+import robot_smach_states.util.designators as ds
+from robot_smach_states import Grab, Inspect, ClassificationResult, NavigateToWaypoint, ForceDrive, SetGripper
+
+try:
+    from typing import List
+except ImportError:
+    pass
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Measure the time it takes to grasp an entity when starting from a given pose. "
+                                                 "After the grasping is done, go back to the waypoint, "
+                                                 "turn around and drop the item, "
+                                                 "then turn around once more to be in the start configuration again")
+    parser.add_argument("cls", type=str, help="class of entity to grasp, eg. 'coke'")
+    parser.add_argument("support", type=str, help="ID of entity to grasp FROM ('grasp' entity is on-top-of this 'support' entity), eg. 'cabinet'")
```

Comment:
I'd say we need a statistically relevant test. So grab one item from the cabinet, kitchen table, desk, ... etc. (maybe up to 10) and compute that duration. Then we know if we really sped things up. I guess we need a default list of entities as well. Maybe even add it to the script: print something like "make sure you have placed entities on x, y, z and z, press enter to continue"

Reply:
Yes, some configuration file for a batch of these tests is to be added.